integration/docker/csi/alluxio/nodeserver.go

```diff
@@ -12,28 +12,51 @@
 package alluxio
 
 import (
+	"fmt"
+	"io/ioutil"
 	"os"
 	"os/exec"
 	"strings"
+	"sync"
+	"time"
 
 	"github.com/container-storage-interface/spec/lib/go/csi"
 	"github.com/golang/glog"
 	csicommon "github.com/kubernetes-csi/drivers/pkg/csi-common"
 	"golang.org/x/net/context"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
 	mount "k8s.io/mount-utils"
 )
 
 type nodeServer struct {
-	nodeId string
+	client kubernetes.Clientset
 	*csicommon.DefaultNodeServer
+	nodeId string
+	mutex  sync.Mutex
 }
 
+/*
+ * If mount in nodeserver, start fuse process
+ * If mount in seperate pod, mount bind global mount point to pod mount point
+ */
 func (ns *nodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublishVolumeRequest) (*csi.NodePublishVolumeResponse, error) {
+	if req.GetVolumeContext()["mountInPod"] == "false" {
+		glog.V(4).Infoln("Will start Fuse process in csi-node-plugin pod.")
+		return nodePublishVolumeMountProcess(req)
+	}
+	glog.V(4).Infoln("Will start Fuse process in alluxio-fuse pod.")
+	return nodePublishVolumeMountPod(req)
+}
+
+func nodePublishVolumeMountProcess(req *csi.NodePublishVolumeRequest) (*csi.NodePublishVolumeResponse, error) {
```

Comment:
Make it as a member function of `nodeServer`?

Reply:
Sorry yes that would be better

Reply:
Why should we do it? We are not using the `nodeServer` in it

Reply:
You can ignore this if not make sense to you